providers/flagd/pkg/service/service.go

```diff
@@ -190,21 +197,46 @@ func handleError(err error) openfeature.ResolutionError {
 }
 
 // EventStream emits received events on the given channel
-func (s *Service) EventStream(ctx context.Context, eventChan chan<- *schemaV1.EventStreamResponse, errorChan chan<- error) {
+func (s *Service) EventStream(
+	ctx context.Context, eventChan chan<- *schemaV1.EventStreamResponse, maxAttempts int, errChan chan<- error,
+) {
+	delay := s.baseRetryDelay
+	var err error
+	for i := 1; i <= maxAttempts; i++ {
+		log.Infof("attempt %d at connecting to event stream", i)
+		i, err = s.eventStream(ctx, eventChan, i, maxAttempts)
+		if i == 1 {
+			delay = s.baseRetryDelay // reset delay if the connection was successful before failing
+		}
+		if err != nil && i <= maxAttempts {
+			delay = 2 * delay
```

Comment:
some form of offset should be used here, especially if a future deployment model is n:1

Reply:
Ins't that what `delay = 2 * delay` is? This would be a exponential (order 2) back-off no?

Reply:
As far as I'm aware its best practice to have an exponential back off with some degree of randomisation, otherwise there is potential for all deployments to be in sync, so if the reason for the failure is throttling then there is no chance for recovery. (this may be overkill of course)